### ISSUE TITLE
Unifile throws error instead of rejecting Promise

### ISSUE THREAD

**Issue by lexoyo** [silexlabs/unifile#148]
Hi @JbIPS 

I found cases where unifile throws error instead of a "rejected" Promise

For example when you call `unifile.readdir` with a service which does not exist, it throws the error `Error: Unknown connector: fs ` instead of returning Promise.reject with this same error

Am I mistaken? Is this the intended behaviour?

Thx!

**Comment by JbIPS** [comment]
You're right, it shouldn't. I think it's because it fails before the async call but it should be wrapped in a Promise for consistency.

**Comment by lexoyo** [comment]
Same for writeFile, cf CloudExplorer2 router.js file, there is a comment about that
